### PR TITLE
Enable pipeline engine in 2.2 in default

### PR DESCRIPTION
**Pull request by satanson**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] others

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Review by ZiheLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -117,7 +117,7 @@
     // for chunk arrival and blocked on receive queues of its exchange node, so among
     // threads allocated for a query in the old execution engine, only small number of
     // them do the real work on core.
-    public static final String ENABLE_PIPELINE_ENGINE = "enable_pipeline_engine";
+    public static final String ENABLE_PIPELINE = "enable_pipeline";
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:120]: should keep compatibility.

**Comment by kangkaisen** [comment]
should pass the daily test firstly

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 1 above)
Comment (satanson) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:120]: Unrecognized keys in json are ignored, so it is compatible.
```java
    public void replayFromJson(String json) throws IOException {
        JSONObject root = new JSONObject(json);
        try {
            for (Field field : SessionVariable.class.getDeclaredFields()) {
                VarAttr attr = field.getAnnotation(VarAttr.class);
                if (attr == null) {
                    continue;
                }

                **if (!root.has(attr.name())) {
                    continue;
                }**

                switch (field.getType().getSimpleName()) {
                    case "boolean":
                        field.set(this, root.getBoolean(attr.name()));
                        break;
                    case "int":
                        field.set(this, root.getInt(attr.name()));
                        break;
                    case "long":
                        field.set(this, root.getLong(attr.name()));
                        break;
                    case "float":
                        field.set(this, root.getFloat(attr.name()));
                        break;
                    case "double":
                        field.set(this, root.getDouble(attr.name()));
                        break;
                    case "String":
                        field.set(this, root.getString(attr.name()));
                        break;
                    default:
                        // Unsupported type variable.
                        throw new IOException("invalid type: " + field.getType().getSimpleName());
                }
            }
        } catch (Exception e) {
            LOG.warn("failed to read session variable: {}", e.getMessage());
        }
    }
```

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 2]
```diff
@@ -189,8 +191,12 @@
 
     public static final String SINGLE_NODE_EXEC_PLAN = "single_node_exec_plan";
 
-    @VariableMgr.VarAttr(name = ENABLE_PIPELINE_ENGINE)
-    private boolean enablePipelineEngine = false;
+    @VariableMgr.VarAttr(name = ENABLE_PIPELINE)
+    private boolean enablePipelineEngine = true;
+
+    @VariableMgr.VarAttr(name = ENABLE_PIPELINE_ENGINE, flag = VariableMgr.INVISIBLE)
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:197]: why not use alias annotation?

**Review by kangkaisen — DISMISSED** [review]

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest.

**Comment by kangkaisen** [comment]
Should merge after daily test pass

**Review by stdpain — DISMISSED** [review]

**Review by murphyatwork — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by satanson** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Review by stdpain — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
@mergify  backport branch-2.2

**Comment by satanson** [comment]
@Mergifyio backport branch-2.2

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|

**Pull request merged**